### PR TITLE
[Visual] Introduce TrailRenderer

### PR DESCRIPTION
**Pull request by alxbilger**
A tool to visualize the trajectory of a particle

particles

pendulum

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by damienmarchal** [comment]
A must have !

**Comment by epernod** [comment]
🎆

**Review by fredroy — APPROVED** [review]

**Comment by fredroy** [comment]
[ci-build][with-scene-tests]

File: Sofa/Component/Visual/src/sofa/component/visual/TrailRenderer.inl [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    const auto stateLifeCycle = vparams->drawTool()->makeStateLifeCycle();
+
+    const auto& color = d_color.getValue();
+    const auto thickness = d_thickness.getValue();
+
+    for (auto& queue : m_trail)
+    {
+        vparams->drawTool()->drawLineStrip(queue, thickness, color);
```
Comment (damienmarchal) [inline Sofa/Component/Visual/src/sofa/component/visual/TrailRenderer.inl]: vparams->drawTool() could be moved out of the loop.

Comment (alxbilger) [reply Sofa/Component/Visual/src/sofa/component/visual/TrailRenderer.inl]: done. Thanks

**Review by epernod — APPROVED** [review]

**Pull request merged**